test-suite/testkit/src/lib.rs

```diff
@@ -430,10 +430,15 @@ impl TestKit {
         // `create_block_with_transactions()` will panic.
         let snapshot = self.snapshot();
         let schema = snapshot.for_core();
-        let uncommitted_txs = transactions.into_iter().filter(|tx| {
-            !schema.transactions().contains(&tx.object_hash())
-                || schema.transactions_pool().contains(&tx.object_hash())
-        });
+        let uncommitted_txs: Vec<_> = transactions
+            .into_iter()
+            .filter(|tx| {
+                self.check_tx(&tx);
```

Comment:
Nit / future work: Could make sense to add a couple `TestKit` tests covering an invalid transaction in `create_block_*` arguments, and in `add_tx`.